provisioning/install-dune.sh

```diff
@@ -1,72 +1,86 @@
 fi
 
 if [ ! -d "dune-grid/" ]; then
-    git clone --branch v2.8.0 --depth=1 https://gitlab.dune-project.org/core/dune-grid.git
+    git clone --depth 1 --branch v2.9.1 https://gitlab.dune-project.org/core/dune-grid.git
 fi
 
 if [ ! -d "dune-geometry/" ]; then
-    git clone --branch v2.8.0 --depth=1 https://gitlab.dune-project.org/core/dune-geometry.git
+    git clone --depth 1 --branch v2.9.1 https://gitlab.dune-project.org/core/dune-geometry.git
 fi
 
-if [ ! -d "dune-functions/" ]; then
-    git clone --branch v2.8.0 --depth=1 https://gitlab.dune-project.org/staging/dune-functions.git
+if [ ! -d "dune-foamgrid/" ]; then
+    # The missing v in the tag in this module originates from the project itself
+    git clone --depth 1 --branch 2.9.1 https://gitlab.dune-project.org/extensions/dune-foamgrid.git
 fi
 
-if [ ! -d "dune-uggrid/" ]; then
-    git clone --branch v2.8.0 --depth=1 https://gitlab.dune-project.org/staging/dune-uggrid.git
+if [ ! -d "dune-functions/" ]; then
+    git clone --depth 1 --branch v2.9.1 https://gitlab.dune-project.org/staging/dune-functions.git
 fi
 
 if [ ! -d "dune-typetree/" ]; then
-    git clone --branch v2.8.0 --depth=1 https://gitlab.dune-project.org/staging/dune-typetree.git
+    git clone --depth 1 --branch v2.9.1 https://gitlab.dune-project.org/staging/dune-typetree.git
 fi
 
-if [ ! -d "dune-foamgrid/" ]; then
-    git clone --branch releases/2.8 --depth=1 https://gitlab.dune-project.org/extensions/dune-foamgrid.git
+if [ ! -d "dune-uggrid/" ]; then
+    git clone --depth 1 --branch v2.9.1 https://gitlab.dune-project.org/staging/dune-uggrid.git
 fi
 
-# Get the dune-elastodynamics module (solid solver)
+# Get the dune-elastodynamics module (solid solver for the plain dune adapter)
 if [ ! -d "dune-elastodynamics/" ]; then
-    git clone --branch master --depth=1 https://github.com/maxfirmbach/dune-elastodynamics.git
+    git clone --depth 1 --branch master https://github.com/maxfirmbach/dune-elastodynamics.git
 fi
 (
     cd dune-elastodynamics
     git pull
 )
 
-# Get the DUNE-preCICE adapter
+# Get the plain DUNE-preCICE adapter
 if [ ! -d "dune-adapter/" ]; then
-    git clone --branch main --depth=1 https://github.com/precice/dune-adapter.git
+    git clone --depth 1 --branch main https://github.com/precice/dune-adapter.git
 fi
 (
     cd dune-adapter/dune-precice
     git pull
 )
 
-# Build all the DUNE and DUNE-preCICE related modules
-DUNE_CONTROL_PATH=~/dune ./dune-common/bin/dunecontrol all
+./dune-common/bin/dunecontrol all
+
+# Get DuMuX and the DuMuX-preCICE adapter
+if [ ! -d "dumux/" ]; then
+    git clone --depth 1 --branch releases/3.8 https://git.iws.uni-stuttgart.de/dumux-repositories/dumux.git
+fi
+
+if [ ! -d "dumux-adapter/" ]; then
+    git clone  --depth 1 --branch v2.0.0 https://github.com/precice/dumux-adapter.git
+fi
+
+# Build the DuMuX-preCICE adapter
+./dune-common/bin/dunecontrol --opts=dumux/cmake.opts --module=dumux-precice all
```

Comment:
Note that this builds the module `dumux-precice` and its recursive dependencies, which in this case is almost all the modules. Due to different CMake settings it will actually reconfigure and rebuild them and leave the installation in a questionable state.
You can build only these two modules by using the `--only=` option in two separate runs, but I would recommend building everything together with your preferred set of CMake options. As far as I can tell both default and those of DuMux should work, using DUNE's default settings is probably safer.

Reply:
Thanks! Where would you get the default DUNE `cmake.opts` from? I understand that only DuMux is defining such a file.

If I skip the `--opts`, is that enough?

I still want to build modules step by step, as this helps a lot with debugging and rebuilding.

Reply:
Yes, skipping `--opts` is enough, just like the call to dunecontrol before cloning DuMux and the adapter.

I don't understand the "rebuilding" aspect, but debugging I can definitely see. If you are using the same options, then it will just do a small run over the already built modules and since the added modules are not dependencies to those already built I see no further issue.

Reply:
By "rebuilding" I essentially mean the same as "development and debugging".
We can execute these scripts either from scratch, or again after reaching some intermediate state. This is why most of the `if [ -d blah/ ]; then` is there.

I understand that the argument is weak for not building everything at once, but I just find that it helps me move things around, enable/disable, understand which step some error originates from, refine some steps, add documentation on restrictions, etc. Not the most elegant way, but I think more easy to understand sometimes.

Reply:
To be clear, I simply didn't understand what you meant by "rebuilding". Seems good to me.